src/partners/thorchain.ts

```diff
@@ -176,12 +184,25 @@ export const thorchain: PartnerPlugin = {
 }
 
 export function processThorchainTx(rawTx: unknown): StandardTx {
-  const tx = asThorchainTx(rawTx)
+  const rawTxs: unknown[] = Array.isArray(rawTx) ? rawTx : [rawTx]
+  const txs = asArray(asThorchainTx)(rawTxs)
+  const tx = txs.shift()
+
+  if (tx == null) {
+    throw new Error('Missing rawTx')
+  }
+
   const srcAsset = tx.in[0].coins[0].asset
   const [chainAsset] = srcAsset.split('-')
   const [, asset] = chainAsset.split('.')
 
-  const depositAmount = Number(tx.in[0].coins[0].amount) / THORCHAIN_MULTIPLIER
+  const extraDepositAmount = txs.reduce((sum, tx) => {
+    const depositAmount =
+      Number(tx.in[0].coins[0].amount) / THORCHAIN_MULTIPLIER
+    return sum + depositAmount
+  }, 0)
+  const depositAmount =
```

Comment:
You're duplicating code here unnecessarily. Just do `const tx = txs[0]` above on line 189, then do the reduce over all the txs. No need for the extraDepositAmount.